library/gcse/src/gcse_api.e

```diff
@@ -0,0 +1,243 @@
+							end
+								-- Current Page
+							if attached {GCSE_PAGE} query_page (request_key, jqueries) as l_page then
+								l_gcse_response.set_current_page (l_page)
+							end
+								-- Previous Page
+							if attached {GCSE_PAGE} query_page (previous_page_key, jqueries) as l_page then
+								l_gcse_response.set_previous_page (l_page)
+							end
+						end
+						if attached {JSON_ARRAY} jv.item (items_key) as jitems then
+							across jitems as ic loop
+								if attached{JSON_OBJECT} ic.item as j_item  then
+									l_gcse_response.add_item (item (j_item))
+								end
+							end
+						end
+
+					end
+				else
+					put_error (l_response.status.out)
+				end
+			else
+				put_error ("unknown")
+			end
+			last_result := l_gcse_response
+		end
+
+feature {NONE} -- REST API
+
+	get: detachable RESPONSE
+			-- Reading Data.
+		local
+			l_request: REQUEST
+		do
+			create l_request.make ("GET", new_uri)
+			Result := l_request.execute
+		end
+
+feature {NONE} -- Implementation
+
+	new_uri: STRING_8
+			-- new uri (BaseUri?key=secret_value&cx=a_cx_id&q=a_query
+			-- ?key=INSERT_YOUR_API_KEY&cx=017576662512468239146:omuauf_lfve&q=lectures
+			-- full teamplte BaseUri?q={searchTerms}&num={count?}&start={startIndex?}&lr={language?}&
+			-- safe={safe?}&cx={cx?}&cref={cref?}&sort={sort?}&filter={filter?}&gl={gl?}&cr={cr?}&
+			-- googlehost={googleHost?}&c2coff={disableCnTwTranslation?}&hq={hq?}&hl={hl?}&siteSearch={siteSearch?}&
+			-- siteSearchFilter={siteSearchFilter?}&exactTerms={exactTerms?}&excludeTerms={excludeTerms?}&linkSite={linkSite?}&
+			-- orTerms={orTerms?}&relatedSite={relatedSite?}&dateRestrict={dateRestrict?}&lowRange={lowRange?}&highRange={highRange?}&
+			-- searchType={searchType}&fileType={fileType?}&rights={rights?}&imgSize={imgSize?}&imgType={imgType?}&
+			-- imgColorType={imgColorType?}&imgDominantColor={imgDominantColor?}&alt=json"
+
+		do
+			create Result.make_from_string (base_uri)
+			Result.append ("?key=")
+			Result.append (query_paremeter.secret)
+			Result.append ("&cx=")
+			Result.append (query_paremeter.cx)
+			Result.append ("&q=")
+			Result.append (query_paremeter.query)
+				-- num
+			if attached query_paremeter.num as l_num then
+				Result.append ("&num=")
+				Result.append (l_num)
+			end
+			if attached query_paremeter.start as l_start then
+				Result.append ("&start=")
+				Result.append (l_start)
+			end
+		end
+
+	put_error (a_code: READABLE_STRING_GENERAL)
+			-- put error with code `a_code'.
+		local
+			l_errors: like errors
+			utf: UTF_CONVERTER
+		do
+			l_errors := errors
+			if l_errors = Void then
+				create {ARRAYED_LIST [STRING]} l_errors.make (1)
+				errors := l_errors
+			end
+			l_errors.force (utf.utf_32_string_to_utf_8_string_8 (a_code))
+		end
+
+	item (a_item: JSON_OBJECT): GCSE_PAGE_ITEM
+			-- Google Result Metadata Item.
+		do
+			create Result
+			if attached {JSON_STRING} a_item.item (create {JSON_STRING}.make_from_string ("kind")) as l_kind then
```

Comment:
you can also write

    a_item.item ("kind")
there is a conversion from string_8 to json_string object

it is the same at the end, but easier to read.